tests/scalers/aws/aws_cloudwatch_min_value_null_values/aws_cloudwatch_min_value_null_values_test.go

```diff
@@ -0,0 +1,242 @@
+        ports:
+        - containerPort: 80
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  maxReplicaCount: 2
+  minReplicaCount: 0
+  cooldownPeriod: 1
+  triggers:
+    - type: aws-cloudwatch
+      authenticationRef:
+        name: keda-trigger-auth-aws-credentials
+      metadata:
+        awsRegion: {{.AwsRegion}}
+        namespace: {{.CloudWatchMetricNamespace}}
+        dimensionName: {{.CloudWatchMetricDimensionName}}
+        dimensionValue: {{.CloudWatchMetricDimensionValue}}
+        metricName: {{.CloudWatchMetricName}}
+        targetMetricValue: "1"
+        minMetricValue: "1"
+        metricCollectionTime: "120"
+        metricStatPeriod: "60"
+`
+)
+
+var (
+	testNamespace                  = fmt.Sprintf("%s-ns", testName)
+	deploymentName                 = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName               = fmt.Sprintf("%s-so", testName)
+	secretName                     = fmt.Sprintf("%s-secret", testName)
+	cloudwatchMetricName           = fmt.Sprintf("cw-%d", GetRandomNumber())
+	awsAccessKeyID                 = os.Getenv("TF_AWS_ACCESS_KEY")
+	awsSecretAccessKey             = os.Getenv("TF_AWS_SECRET_KEY")
+	awsRegion                      = os.Getenv("TF_AWS_REGION")
+	cloudwatchMetricNamespace      = "DoesNotExist"
+	cloudwatchMetricDimensionName  = "dimensionName"
+	cloudwatchMetricDimensionValue = "dimensionValue"
+	minReplicaCount                = 0
+	minMetricValueReplicaCount     = 1
+)
+
+func TestCloudWatchScalerWithMinValueWhenNullValues(t *testing.T) {
+	// setup cloudwatch
+	cloudwatchClient := createCloudWatchClient()
+
+	// check that the metric in question is not already present, and is returning
+	// an empty set of values.
+	checkCloudWatchCustomMetric(t, cloudwatchClient)
+
+	// Create kubernetes resources
+	kc := GetKubernetesClient(t)
+	kedaClient := GetKedaKubernetesClient(t)
+	data, templates := getTemplateData()
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+	defer DeleteKubernetesResources(t, testNamespace, data, templates)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minReplicaCount, 60, 1),
+		"replica count should be %d after 1 minute", minReplicaCount)
+
+	// check that the scaledobject is in paused state
+	FailIfScaledObjectStatusNotReachedWithTimeout(t, kedaClient, testNamespace, scaledObjectName, 2*time.Minute, v1alpha1Api.ConditionPaused)
+
+	// check that the deployment scaled up to the minMetricValueReplicaCount
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, minMetricValueReplicaCount, 60, 1),
+		"replica count should be %d after 1 minute", minMetricValueReplicaCount)
+}
+
+func createCloudWatchClient() *cloudwatch.Client {
```

Comment:
I've not deferred from the structure of the other e2e tests for AWS, but there is some duplication, for example here, if you wanted me to create a test package and put the shared functionality in there - let me know whats best.

Reply:
yeah, please do that 🙏

Reply:
Added an initial set of shared functions for the e2e tests I added, and if your happy with this, over the next few weeks I'll present smaller PRs to refactor the remaining aws e2e tests into shared functions.